pkgs/tools/networking/phantomsocks/default.nix

```diff
@@ -0,0 +1,25 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "phantomsocks";
+  version = "unstable-2023-03-16";
+
+  src = fetchFromGitHub {
+    owner = "macronut";
+    repo = pname;
+    rev = "7cc559b152f9d3456e630eaf9f2f689970ae238d";
+    hash = "sha256-+d/cXMZAQT9QJVSlLr4Tp19HpbRMl8jYUO8EPZFVarU=";
+  };
+
+  vendorHash = "sha256-c0NQfZuMMWz1ASwFBcpMNjxZwXLo++gMYBiNgvT8ZLQ=";
+
+  meta = with lib;{
+    homepage = "https://github.com/macronut/phantomsocks";
+    description = "A cross-platform proxy client/server for Linux/Windows/macOS";
+    license = licenses.lgpl3;
```

Comment:
One more nitpick: `lgpl3` is deprecated, please choose `lgpl3Plus` or `lgpl3Only` after carefully inspecting the upstream license.